### ISSUE TITLE
TensorRT LLM backend.

### ISSUE THREAD

**Issue by Ph0rk0z** [oobabooga/text-generation-webui#4332]
So a new inference engine has entered the arena. Claims good speeds from nvidia HW.

Looks like you can call it from python: https://github.com/NVIDIA/TensorRT-LLM/tree/release/0.5.0/examples/llama

Might make for faster AWQ models at least. Sadly it's volta and up only and uses triton. Has anyone tried it yet?

**Comment by berkut1** [comment]
Is the driver version 531.79 or older supported? Or did we lose some performance after 531.79 and are trying to regain it using a new backend?

**Comment by Ph0rk0z** [comment]
I can run 12 so I must be over the hump. I don't want their funky memory driver.

**Comment by Pevernow** [comment]
+1

**Comment by m5kro** [comment]
+1

**Comment by lamkhatinh** [comment]
Great feature. Vote it!

**Comment by rafalohaki** [comment]
approve

**Comment by taoxunqiang** [comment]
> Great feature. Vote it!

+1

**Comment by FAChenier** [comment]
I noticed a "tensor core" feature in the llama.cpp model settings, is it this or completely unrelated?

If not, bump. Started using AUTOMATIC1111's imagen webui which has an extension *made by* Nvidia to add the diffuser version of this and it has an incredible impact on RTX cards. Went from going over VRAM and taking over 20 seconds to staying in VRAM, with headroom, and taking at most 6 seconds.

If it can have a similar impact on LLM then maybe larger models will be available on consumer hardware at reasonable speeds lol, might be optimistic though.

**Comment by Ph0rk0z** [comment]
No, tensor core is just a different kernel, for me it's slower. I used diffusers in SD-next and the speed is about the same. Also compiling the model with the old tensorrt they had for SD didn't yield any performance. Haven't fully explored the new model optimizations as they require removing xformers. 

This one is different though, its an optimizer and runtime.

**Comment by FAChenier** [comment]
> No, tensor core is just a different kernel, for me it's slower. I used diffusers in SD-next and the speed is about the same. Also compiling the model with the old tensorrt they had for SD didn't yield any performance. Haven't fully explored the new model compiling optimizations as they require removing xformers. 
> 
> This one is different though, its an optimizer and runtime.

Thank you for the detailed response.

Whatever the outcome is, all I'd ever want is for my GPU to be fully used. But either way, I'll read into all of this, I'm quite new to this whole thing. Thanks again.

As for diffusers, idk the exact details but the tensor add-on had a very noticeable impact, I can give more details if curious.
